File: mobile/spag_app/lib/features/admin/models/service_status_log.dart
class ServiceStatusLog {
  final int id;
  final int serviceId;
  final String? oldStatus;
  final String newStatus;
  final String changedAt;
  final int? changedBy;

  ServiceStatusLog({
    required this.id,
    required this.serviceId,
    this.oldStatus,
    required this.newStatus,
    required this.changedAt,
    this.changedBy,
  });

  factory ServiceStatusLog.fromJson(Map<String, dynamic> json) {
    return ServiceStatusLog(
      id: json['id'] as int,
      serviceId: json['service_id'] as int,
      oldStatus: json['old_status'] as String?,
      newStatus: json['new_status'] as String? ?? '',
      changedAt: json['changed_at']?.toString() ?? '',
      changedBy: json['changed_by'] as int?,
    );
  }
}

File: backend/app/models/service_status_log.py
from sqlalchemy import Column, Integer, String, ForeignKey, DateTime
from sqlalchemy.sql import func
from app.database import Base


class ServiceStatusLog(Base):
    __tablename__ = "service_status_logs"

    id = Column(Integer, primary_key=True, index=True)

    service_id = Column(
        Integer,
        ForeignKey("service_history.id", ondelete="CASCADE"),
        nullable=False,
    )

    old_status = Column(String, nullable=True)
    new_status = Column(String, nullable=False)

    changed_by = Column(
        Integer,
        ForeignKey("users.id"),
        nullable=True,
    )

    # changed_by_role = Column(String, nullable=True)

    changed_at = Column(
        DateTime(timezone=True),
        server_default=func.now(),
        nullable=False,
    )
